src/ngraph/attribute_visitor.hpp

```diff
@@ -0,0 +1,46 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <string>
+
+#include "ngraph/partial_shape.hpp"
+#include "ngraph/type/element_type.hpp"
+
+namespace ngraph
+{
+    class VisitorAdapter;
+
+    /// \brief Visits the attributes of a node.
+    ///
+    /// Attributes are the values set when building a graph which are not
+    /// computed as the graph executes. Values computed from the graph topology and attributes
+    /// during compilation are not attributes.
+    class AttributeVisitor
+    {
+    public:
+        virtual ~AttributeVisitor() {}
+        virtual void on_attribute(const std::string& name, std::string& value) = 0;
+        virtual void on_attribute(const std::string& name, element::Type& value) = 0;
+        virtual void on_attribute(const std::string& name, PartialShape& value) = 0;
+        virtual void on_attribute(const std::string& name, bool& value) = 0;
+        virtual void on_attribute(const std::string& name, int64_t& value) = 0;
+        virtual void on_attribute(const std::string& name, uint64_t& value) = 0;
+        virtual void on_attribute(const std::string& name,
```

Comment:
yes, as needed.

Reply:
But from my point of view we must implement these methods for all basic types at the same time or have only one method with VisitorAdapter for all type. Because in another case if we didn't implement (for example) uint8_t, but user wants to serialise/deserialise this type in own custom op, he will have to write a special adapter in order to work with it.

It is inconvenient.